sqlitecluster/SQLiteNode.cpp

```diff
@@ -326,81 +326,52 @@ bool SQLiteNode::_isNothingBlockingShutdown() const {
     return true;
 }
 
-bool SQLiteNode::shutdownComplete() {
+bool SQLiteNode::shutdownComplete() const {
+    shared_lock<decltype(_stateMutex)> sharedLock(_stateMutex);
+
     // First even see if we're shutting down
-    if (!_gracefulShutdown())
+    if (!_gracefulShutdown()) {
         return false;
+    }
 
     // Next, see if we're timing out the graceful shutdown and killing non-gracefully
     if (_shutdownTimeout.ringing()) {
         SWARN("Graceful shutdown timed out, killing non gracefully.");
-        auto lock = _escalatedCommandMap.scopedLock();
-        if (_escalatedCommandMap.size()) {
-            SWARN("Abandoned " << _escalatedCommandMap.size() << " escalated commands.");
-            for (auto& commandPair : _escalatedCommandMap) {
-                commandPair.second->response.methodLine = "500 Abandoned";
-                commandPair.second->complete = true;
-                _server.acceptCommand(move(commandPair.second), false);
-            }
-            _escalatedCommandMap.clear();
-        }
-        _changeState(SEARCHING);
```

Comment:
Do we need to keep this line?

Reply:
No, because when this returns we fall out of this main loop and this becomes irrelevant:
https://github.com/Expensify/Bedrock/blob/6d84191ca0afdb55ad26f22957cfb91e9a547df1/BedrockServer.cpp#L660

Leaving it in actually caused some weirdness in one of the intermediate commits in this branch though I can't remember exactly what it was now. I did investigate that it was safe to remove.

Reply:
Figured out what the weirdness was: We made this method `const` so it can no longer change state, but this way we can shared_lock it instead of unique_lock.

And because as soon as we would have made this change and then returned `true`, it becomes irrelevant anyway, the node is getting cleaned up and destroyed.